### ISSUE TITLE
icon images that can be used after uploaded is hard limited to 60

### ISSUE THREAD

**Issue by mahnkej** [unfoldedcircle/feature-and-bug-tracker#52]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

It seems that the maximum number of images that can be uploaded is hard limited to 60. This is because each additional upload is no longer displayed and the number of images no longer increases.

<img width="2059" alt="Bildschirmfoto 2023-09-24 um 17 34 56" src="https://github.com/unfoldedcircle/bug-reports/assets/49618399/4fb15ab8-23cb-4196-a972-d25a52dd50d0">
<img width="2059" alt="Bildschirmfoto 2023-09-24 um 17 39 27" src="https://github.com/unfoldedcircle/bug-reports/assets/49618399/c6123ad9-d39e-41b8-8b03-196930a7ad45">
<img width="2059" alt="Bildschirmfoto 2023-09-24 um 17 39 38" src="https://github.com/unfoldedcircle/bug-reports/assets/49618399/3fa891bc-161d-4834-9cb9-c328d25c764b">

### How to Reproduce

1. Upload 60 images to the Icon folder
2. Upload the 61 images to the folder Icon
3. the 61 pictures are not visible in the folder
4. The 61st image cannot be found via the search option either. However, an error message is returned if the same image is uploaded again.

### Expected behavior

Since it is possible to upload your own icons, the value should be set realistically. If the default is to control the memory requirement, I would suggest that everyone can set the value themselves and receive a warning if the value is set too high. Alternatively, set a fixed value at 500. 

### System version

1.2.1

### What part of the system affected by the problem?

Core

### Additional context

2023-09-24 15:39:35.839044 +00:00	core	WARN	Error encountered while processing the incoming HTTP request: AlreadyExists("/opt/uc/data/data/upload/media/Icon/go-to.png")
2023-09-24 15:28:16.254845 +00:00	core	WARN	Error encountered while processing the incoming HTTP request: AlreadyExists("/opt/uc/data/data/upload/media/Icon/go-to.png")
2023-09-24 15:27:34.360607 +00:00	core	NOTICE	Contacting OTA server for dock update check
2023-09-24 15:27:34.212990 +00:00	core	NOTICE	Check OTA server for dock firmware update: UC-Dock-8C4B14AA754C
2023-09-24 15:27:33.821176 +00:00	core	NOTICE	Contacting OTA server for dock update check
2023-09-24 15:27:33.671302 +00:00	core	NOTICE	Check OTA server for dock firmware update: UC-Dock-8C4B14AA761C
2023-09-24 15:25:58.541837 +00:00	core	WARN	Error encountered while processing the incoming HTTP request: AlreadyExists("/opt/uc/data/data/upload/media/Icon/go-to.png")
2023-09-24 12:26:35.807704 +00:00	core	NOTICE	Contacting OTA server for dock update check
2023-09-24 12:26:35.657456 +00:00	core	NOTICE	Check OTA server for dock firmware update: UC-Dock-8C4B14AA754C
2023-09-24 12:26:35.385342 +00:00	core	NOTICE	Contacting OTA server for dock update check
2023-09-24 12:26:35.229338 +00:00	core	NOTICE	Check OTA server for dock firmware update: UC-Dock-8C4B14AA761C

**Comment by zehnm** [comment]
I can confirm that this is a bug in the web-configurator. 
Instead of retrieving all available icons, only the first 50 are retrieved and then again the first 10. Test with 52 icons:
1. First page: `GET /api/resources/Icon?limit=50&page=1`
   Response headers: 
```
Pagination-Count: 52
Pagination-Limit: 50
Pagination-Page: 1
```
2. Instead of retrieving the next page, the web-configurator calls `GET /api/resources/Icon`!
   Without `limit` and `page` query parameters, the default first 10 results are returned...
3. The combined result is shown in the icon chooser dialog.

Additional information:
- the Core-API currently enforces the following resource file limits:
  -  256 icons
  -  256 TV channel icons (not yet selectable in the web-configurator)
  -  30 background images
  -  50 sound files (not yet implemented in the UI)
- metadata about resource files including restrictions: `GET /api/resources`
  - Icon `max_count` value will be fixed in next released. Should be 256 instead of 100.
- Web-configurator does not yet implement proper paging to retrieve all available information (related to #42)

**Comment by zehnm** [comment]
This is fixed in the next update. The web-configurator is no longer limited to 50 icons (+ 10 duplicates at the end).
The limit will be set to 1024 icons. Hopefully enough for everyone :-)

<img width="785" alt="image" src="https://github.com/unfoldedcircle/feature-and-bug-tracker/assets/17356629/11caa60d-1795-4166-9abc-c98b4d1bd171">


Related features in our backlog:
- Support multi-file upload in the web-configurator.
- Enable tv-channel icons to better separate / organize regular icons from channel icons.

Multi-file upload is already supported in the Core-API and can be used on the command line, in case you'd like to import a bigger collection:

```shell
curl 'http://$REMOTE_URL/api/resources/Icon' \ 
--user web-configurator:$PIN \
--form 'file=@"BBC_Four_logo_2021.png"' \
--form 'file=@"BBC_One_logo_2021.png"' \
--form 'file=@"BBC_Three_2022.png"'
```

**Comment by zehnm** [comment]
Fixed in 1.3.3
